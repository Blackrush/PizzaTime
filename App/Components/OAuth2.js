import React, { Component } from 'react'
import { WebView } from 'react-native'


export class Login extends Component {
  static propTypes = {
    style: React.PropTypes.any,

    api: React.PropTypes.shape({
      authorizeUri: React.PropTypes.string,
      isAppLoginRedirection: React.PropTypes.func,
      appLogin: React.PropTypes.func,
    }),

    onStartLogin: React.PropTypes.func,
    onLoginSuccess: React.PropTypes.func,
    onLoginError: React.PropTypes.func,
  }

  render() {
    return (
      <WebView style={this.props.style} source={{uri: this.props.api.authorizeUri}}
               javaScriptEnabled
               domStorageEnabled
               startInLoadingState
               onShouldStartLoadWithRequest={this._onShouldStartLoadWithRequest.bind(this)}
      />
    )
  }

  _onShouldStartLoadWithRequest = (event) => {
    if (this.props.api.isAppLoginRedirection(event.url)) {
      this.props.onStartLogin();

      this.props.api.appLogin(event.url)
      .then((resp) => {
        this.props.onLoginSuccess(resp)
      })
      .catch((err) => {
        this.props.onLoginError(err)
      })

      return false
    }
    return true
  }
}