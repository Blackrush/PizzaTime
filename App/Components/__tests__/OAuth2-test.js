import React from 'react'
import * as OAuth2 from '../OAuth2'
import { create as render } from 'react-test-renderer'

jest.mock('WebView', () => 'WebView')

test('OAuth2.Login renders correctly', () => {
  const api = {
    authorizeUri: 'mock',
    isAppLoginRedirection() {},
    appLogin() {},
  }

  const tree = render(
    <OAuth2.Login style={{marginTop: 64}}
                  api={api}
                  onStartLogin={() => false}
                  onLoginSuccess={() => false}
                  onLoginError={() => false}
    />
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
