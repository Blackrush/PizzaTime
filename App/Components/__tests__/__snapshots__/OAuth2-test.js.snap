exports[`test OAuth2.Login renders correctly 1`] = `
<WebView
  domStorageEnabled={true}
  javaScriptEnabled={true}
  onShouldStartLoadWithRequest={[Function bound ]}
  source={
    Object {
      "uri": "mock"
    }
  }
  startInLoadingState={true}
  style={
    Object {
      "marginTop": 64
    }
  } />
`;
